lib/airbrake-ruby/promise.rb

```diff
@@ -0,0 +1,75 @@
+module Airbrake
+  ##
+  # Represents a simplified promise object (similar to promises found in
+  # JavaScript), which allows chaining callbacks that are executed when the
+  # promise is either resolved or rejected.
+  #
+  # @see https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Promise
```

Comment:
I think we should model API after https://github.com/ruby-concurrency/concurrent-ruby/blob/master/lib/concurrent/promise.rb. E.g. I noticed that Ruby uses resque instead of catch and it makes sense :)